web/js/map-OpenLayers.js

```diff
@@ -710,48 +697,29 @@ OpenLayers.Control.Click = OpenLayers.Class(OpenLayers.Control, {
         // Store pin location in form fields, and check coverage of point
         fixmystreet_update_pin(lonlat);
 
-        // Already did this first time map was clicked, so no need to do it again.
-        if (fixmystreet.page == 'new') {
-            return;
+        // If there are notes to be displayed, add the .with-notes class
+        // to make the sidebar wider.
+        if($('#report-a-problem-sidebar').length){
+          $('#map_box, #map_sidebar').addClass('with-notes');
```

Comment:
Yes to all of this!

I hadn't considered that there might be a way to do it server-side. Can we suck the contents of the sidebar template in, and assign it to a variable, and then check the variable for emptiness to see whether a sidebar has been declared? And then, obviously, print out the variable at the required point.